sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,196 @@
+			if(appExtLessSecurePropertiesFile!=null) {
+				providerURL=appExtLessSecurePropertiesFile.getProperty(PROPERTY_LDAP_SERVER_URL);
+				mcommunityGroup=appExtLessSecurePropertiesFile.getProperty(PROPERTY_AUTH_GROUP);
+			}else {
+				M_log.warn("Failed to load secure application properties from sectionsToolPropsLessSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathLessSecure.properties) is not provided");
+		}
+
+		
+	}
+	 private boolean isEmpty(String value) {
+		return (value == null) || (value.trim().equals(""));
+	}
+	
+	/*
+	 * User is authenticated using cosign and authorized using Ldap. For local development we are enabling
+	 * "testUser" parameter. "testUser" also go through the Ldap authorization process. 
+	 * we have use.test.url configured in the properties file in order to disable usage of "testUser" parameter in PROD.
+	 * 
+	 */
+	private boolean checkForAuthorization(HttpServletRequest request) {
+		M_log.debug("checkLdapForAuthorization(): called");
+     	String remoteUser = request.getRemoteUser();
+     	String testUser = request.getParameter(TEST_USER);
+     	boolean isAuthorized = false;
+		String user=null;
+		
+		String testUserInSession = (String)request.getSession().getAttribute(TEST_USER);
+		
+	    if ( isTestUrlEnabled && testUser != null ) { 
+	    	user=testUser;
+			request.getSession().setAttribute(TEST_USER, testUser);
+		}
+		else if ( isTestUrlEnabled && testUserInSession != null ){
+			user=testUserInSession;
+		} 
+		if  ( !isAuthorized && remoteUser != null ) {
+			user=remoteUser;
+		}
+		isAuthorized=ldapAuthorizationVerification(user); 
+		return isAuthorized;
+		
+		
+	}
+     /*
+      * The Mcommunity group we have is a members-only group is one that only the members of the group can send mail to. 
+      * The group owner can turn this on or off.
+      * More info on Ldap configuration  http://www.itcs.umich.edu/itcsdocs/r1463/attributes-for-ldap.html#group.
+      */
+	private boolean ldapAuthorizationVerification(String user) {
+		M_log.debug("ldapAuthorizationVerification(): called");
+		boolean isMem = false;
+		Hashtable<String,String> env = new Hashtable<String, String>();
+		if(!isEmpty(providerURL) && !isEmpty(mcommunityGroup)) {
+		env.put(Context.INITIAL_CONTEXT_FACTORY, LDAP_CTX_FACTORY);
+		env.put(Context.PROVIDER_URL, providerURL);
+		}else {
+			M_log.error(" [ldap.server.url] or [mcomm.group] properties are not set, review the sectionsToolPropsLessSecure.properties file");
+			return isMem;
+		}
+		try {
+			DirContext dirContext = new InitialDirContext(env);
+			String[] attrIDs = {"member"};
+			SearchControls searchControls = new SearchControls();
+			searchControls.setReturningAttributes(attrIDs);
+			searchControls.setReturningObjFlag(true);
+			searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+			String searchBase = OU_GROUPS;
```

Comment:
Is "ou=Groups" really enough to specify the search base?  What about the rest of the DN, like "dc=umich, dc=edu"?